src/tsd/QueryRpc.java

```diff
@@ -115,14 +122,45 @@ private void handleQuery(final TSDB tsdb, final HttpQuery query) {
       throw new BadRequestException(HttpResponseStatus.BAD_REQUEST, 
           e.getMessage(), data_query.toString(), e);
     }
-    
+
     Query[] tsdbqueries;
     try {
       tsdbqueries = data_query.buildQueries(tsdb);
     } catch(NoSuchUniqueName ex) {
       throw new BadRequestException(ex);
     }
+
     final int nqueries = tsdbqueries.length;
+    if (query.method() == HttpMethod.DELETE) {
```

Comment:
I'd prefer if you break this out into a separate function please.
Also, add a setting in the config class, something like "tsd.http.query.allow_delete" that is set to false by default. We don't want folks deleting on accident :)